src/main/resources/openapi/PrintAPIs.yaml

```diff
@@ -27,6 +27,12 @@ paths:
           type: string
         in: path
         required: true
+      - name: x-correlation-id
```

Comment:
It's useful to document at an API level that the contract accepts a correlation Id. I could not find a global way of doing it so had to touch all the endpoints and add a optional header. I'm open to suggestions if there are better accepted ways of documenting it.

Reply:
This is a tricky one, and I dont know the answer
Yes, we need to document it in the openAPI spec somehow. 100%

Do we need to do it on every endpoint? I don't know. For larger APIs (such as VCA) this will add loads to the spec in terms of line count, but I wonder if its adding much (documenting every endpoint). Even in this small API (Print API is one of our smaller ones) you've had to add it to 8 endpoints
And of course if this is the approach we are taking we would need to make sure that all devs know about this to be able to consistently follow the pattern when adding new endpoints etc

Kinda related, but in the VAC repos (VCA, RCA, print-api, notifications-api, ero-management and admin-api) we made a decision a while back **not** to document standard error response codes such as 401/403/400/404/409/5xx etc. unless they added specific context or value.
IE. we don't document 401/403 because if the API returns this we'd like to think the client devs will know this is an authentication issue. Likewise 404 - we all know what that means so felt it unnecessary (to the point of being distracting noise) to explicitly document on every endpoint. 400's and 409's get documented only if there is some very specific context or business case that is not obvious (eg: you'll get a 409 CONFLICT if your application is not in status X, Y or Z)

Also, as well as documenting that it's a request header that we accept; based on the changes you made to the interceptor, its also returned as a response header. We would want to document that as well, but not in every endpoint definition 🤔 

Any thoughts? @matt-wills-valtech @stalwartsolutions @Neil-Massey @vishalgupta-valtech @adacayi @sentioservices

Reply:
>  I could not find a global way of doing it so had to touch all the endpoints and add a optional header

Me neither, which is a shame. So if we want to avoid adding this to every endpoint, which is completely understandable, then perhaps one "better than nothing" option is to add some info to the description at the start of the spec? We currently have:

```
info:
  title: Print APIs
  version: '1.11.0'
  description: Print APIs
```

Whatever we decide, this description could be considerably better than just "Print APIs"! 😆